components/camel-as2/camel-as2-component/src/test/java/org/apache/camel/component/as2/MendelsonCertLoader.java

```diff
@@ -0,0 +1,183 @@
+package org.apache.camel.component.as2;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.security.KeyManagementException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.spec.InvalidKeySpecException;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.List;
+
+import javax.net.ssl.SSLContext;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.http.conn.ssl.TrustAllStrategy;
+import org.apache.http.ssl.SSLContexts;
+
+/**
+ * That's a utility class for preparing Mendelson-specific certificate chain, private key, ssl context
+ */
+public class MendelsonCertLoader {
```

Comment:
I am not entirely sure we want this file as it is. This is tied to an specific instance of a service. Instead, I think it would be better to make it generic and configurable, so that anyone willing to test with a different instance could do so.

Reply:
I'm not sure that the requested changes are in the scope of the ticket CAMEL-17946. Maybe we need another ticket like "Provide versatile testing harness for AS2 component".
Let me explain the situation. All the Mendelson-related classes are in the test folder, they aren't part of AS2 JAR artifact. My idea was to provide a basic mechanism so that interested persons can verify my results for HTTPS AS2 connection - scientific approach. I din't aim to provide a versatile mechanism. Mendelson is almost the only public resource for testing, it was mentioned in https://github.com/apache/camel/pull/8896 and not by me. It doesn't bring something proprietary into the code base.
Important facts about Mendelson resource. It provides concrete type of private key and keystore. And its certificate chain is degenerated - consists of the only certificate. Furthermore, the same certificate chain is used for encryption and signing.
The only Mendelson test method is @Disabled by default.
It is important to test AS2 connection to non-localhost server due to it's the only way to test the behaviour of the new config param HostnameVerifier.
One other PRO: it was difficult to construct SSLContext from scratch for me. If I could choose, I would prefer to have real-life code examples in the code base.

Reply:
What I am considering here is: if this is disabled, it won't be tested often. If it won't be tested often, the code will rot and it will be upon us (maintainers/committers) to figure out - several months in the future - if it's still relevant or not (and, as things change, certificates get expired, etc) it's one hell of a pain to update and fix things on these obscure technologies. 

 So, by making it generic and configurable ... users themselves can handle this and put more burden on the maintainers.
 
 I am -1 on this, but let's see what others have to say.

Reply:
Since there was no feedback, let me provide more clarity: what I am asking here is to implement something like this: https://github.com/apache/camel/blob/main/components/camel-kafka/src/test/java/org/apache/camel/component/kafka/integration/KafkaConsumerAuthManualTest.java

That is a manual test for connecting to an authenticated Kafka instance. I wrote to try it verify a problem an specific vendor of a managed Kafka service was having. As you can see in the linked test, there's no reference to any specific instance. I'd like this code to apply the same pattern.

Reply:
I still hope to get more comments from other maintainers and contributors, I'm not in a rush. I think we need to calm down and don't take this PR personal.

I can agree that the test class's name for mendelson should be appended with "ManualTest" and not "IT" like now. But I don't understand why I should deliver a semi-prepared test class with a dependency on env variables instead of fully operational and friendly self-sustainable test class which matches all its requirements. Why a person would need to create env vars on his/her machine? They may intersect with existing vars, he/she may forget to remove them after all. That's an invasive approach.

I have another example for ManualTest class:  https://github.com/apache/camel/blob/main/components/camel-cometd/src/test/java/org/apache/camel/component/cometd/CometdProducerConsumerInteractiveAuthenticatedManualTest.java
Lines 73-74 were last edited by you (I don't say that you were the initial author). And in these lines the test class reads a keystore file from a similar resource folder jsse palced in test section of the sources. The test class CometdProducerConsumerInteractiveAuthenticatedManualTest isn't environment aware. All the config variables are part of the test class. Why is it OK to have it like this?

Reply:
What happen if the Mendelson some day will not work anymore? We need a way to be not dependant on this kind of 3rd party stuff. We are trying to do this in multiple components, by adding containerized services to run when needed. This is really not in line with what we are doing. If this will be disabled for a long time, it will be something maintainers will have to deal with in the future. So I would prefer a different solution, so anyone could maintain this piece of software.

Reply:
I'm not sure that disabled by default test can be considered a dependency. It's just good manners: showing a way a person can establish HTTPS AS2 connection to non-localhost server.

If Mendelson goes down, just remove two classes with Mendelson in the name, the test/resource folder mendelson and the file test-server.properties. I will be able to do that myself. No hurt feelings on my side then :)

I think dockerization of AS2 server is beyond the scope of the ticket CAMEL-17946
Besides, what exactly are you going to dockerize? Camel AS2 server component? Both Camel AS2 server and client can have issues which will allow them to communicate, but not to other AS2-compliant server (what if the sources of both client and server suffer from the same issue?). Another point: it's important to test HTTPS connection to non-localhost server - test the new HostnameVerifier field. Will docker allow that?

I'm not sure you have seen my latest commit with moving Mendelson connection settings to a property file as per @davsclaus 's comment. Now a person needs to turn on 2 "knobs" to connect to Mendelson:
- remove/comment @Disabled annotation
- explicitly set Mendelson hostname in the properties.

Please let me know if the latest commit mitigates or even eliminates your concerns. If it dosen't, I will remove all Mendelson-related stuff.